model_one/cli.py

```diff
@@ -1,198 +1,274 @@
-
-        return self.upload_arrays(train_X.tolist(), train_y.tolist(), validate_X.tolist(), validate_y.tolist())
-
-    def _inference(self, method, input):
-        if not input:
-            raise Exception('prompt must not be empty')
-
-        r = requests.get('{}/{}/{}?input={}'.format(_models_api(),
-                         self._id, method, urllib.parse.quote(input)), headers=_get_auth_header())
         if r.status_code != 200:
-            raise Exception(r.text)
+            raise BeyondmlModelException(r.text)
+
         return r.json()
 
-    def generate(self, input):
-        res = self._inference(method='generate', input=input)
-        return res['answer']['responses'][0]['response']
+    @classmethod
+    def create(cls, data: dict) -> dict:
+        return cls._request(*cls.VO["create"], data)
 
-    def classify(self, input):
-        res = self._inference(method='classify', input=input)
-        return res['answer']['scores']
+    @classmethod
+    def models(cls, data: dict) -> dict:
+        return cls._request(*cls.VO["models"], data)
 
-    def status(self):
-        if not self._id:
-            raise Exception('Initialize the model.')
-        r = requests.get('{}/{}/{}'.format(_models_api(),
-                         self._id, 'status'), headers=_get_auth_header())
-        if r.status_code != 200:
-            raise Exception(r.text)
-        self._status = r.json()['status']
-        return r.json()
+    @classmethod
+    def classify(cls, id: str, input: str) -> dict:
+        method, url = cls.V0["classify"]
+
+        return cls._request(method, url.format(id), params={"input": input})
+
+    @classmethod
+    def generate(cls, id: str, input: str) -> dict:
+        method, url = cls.V0["generate"]
+
+        return cls._request(method, url.format(id), params={"input": input})
+
+    @classmethod
+    def upload(cls, id: str) -> dict:
+        method, url = cls.V0["upload"]
 
+        return cls._request(method, url.format(id))
+
+    @classmethod
+    def status(cls, id: str) -> dict:
+        method, url = cls.V0["status"]
+
+        return cls._request(method, url.format(id))
+
+    @classmethod
+    def fit(cls, id: str) -> dict:
+        method, url = cls.V0["fit"]
+
+        return cls._request(method, url.format(id))
+
+
+class BeyondmlModelStatus(Enum, str):
+    READY = "ready"
+    CREATED = "created"
+    DATASETS_LOADED = "datasetsloaded"
+
+
+class BeyondmlModelType(Enum, str):
+    GENERATOR = "generator"
+    CLASSIFIER = "classifier"
+
+
+def inited(m: callable):
+    @wraps(m)
+    def _wrapper(self: BeyondmlModel, *args, **kwargs):
+        if not self.is_inited:
+            raise BeyondmlModelException("Initialize the model")
+        
+        return m(self, *args, **kwargs)
+    
+    return _wrapper
+
+
+class BeyondmlModel():
+    _id: str = None
+    _status: str = None
+    _model_type: str = None
+
+    def __init__(self, model_name: str, status: str, model_type: str, *args, **kwargs):
```

Comment:
I believe that init should not really contain any logic. Just fill in the fields and proceed
*args & **kwargs are used in case those model-dicts contain something besides required fields and allow us to use fancy keyword arguments syntax inside `from_dict` method